bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -88,31 +89,40 @@
         metadataChangeListener = new RegistryChangeListener<Metadata>() {
             @Override
             public void added(final Metadata metadata) {
-                try {
-                    markDirty(itemRegistry.getItem(metadata.getUID().getItemName()));
-                } catch (ItemNotFoundException e) {
-                    logger.debug("Could not found item for metadata {}", metadata);
+                final MetadataKey uid = metadata.getUID();
+                if (uid.getNamespace().equalsIgnoreCase(HomekitAccessoryFactory.METADATA_KEY)) {
```

Comment:
You could reverse and put HomekitAccessoryFactory.METADATA_KEY at left (in case uid.getNamespace() could be null. 

Same elsewhere.

Reply:
agree, done.